cmd/podmanV2/images/search.go

```diff
@@ -0,0 +1,158 @@
+
+	Users can limit the number of results, and filter the output based on certain conditions.`
+
+	// Command: podman search
+	searchCmd = &cobra.Command{
+		Use:     "search [flags] TERM",
+		Short:   "Search registry for image",
+		Long:    searchDescription,
+		PreRunE: preRunE,
+		RunE:    imageSearch,
+		Example: `podman search --filter=is-official --limit 3 alpine
+  podman search registry.fedoraproject.org/  # only works with v2 registries
+  podman search --format "table {{.Index}} {{.Name}}" registry.fedoraproject.org/fedora`,
+	}
+
+	// Command: podman image search
+	imageSearchCmd = &cobra.Command{
+		Use:     searchCmd.Use,
+		Short:   searchCmd.Short,
+		Long:    searchCmd.Long,
+		PreRunE: searchCmd.PreRunE,
+		RunE:    searchCmd.RunE,
+		Example: `podman image search --filter=is-official --limit 3 alpine
+		podman image search registry.fedoraproject.org/  # only works with v2 registries
+		podman image search --format "table {{.Index}} {{.Name}}" registry.fedoraproject.org/fedora`,
+	}
+)
+
+func init() {
+	// pull
+	registry.Commands = append(registry.Commands, registry.CliCommand{
+		Mode:    []entities.EngineMode{entities.ABIMode, entities.TunnelMode},
+		Command: searchCmd,
+	})
+
+	searchCmd.SetHelpTemplate(registry.HelpTemplate())
+	searchCmd.SetUsageTemplate(registry.UsageTemplate())
+	flags := searchCmd.Flags()
+	searchFlags(flags)
+
+	// images pull
+	registry.Commands = append(registry.Commands, registry.CliCommand{
+		Mode:    []entities.EngineMode{entities.ABIMode, entities.TunnelMode},
+		Command: imageSearchCmd,
+		Parent:  imageCmd,
+	})
+
+	imageSearchCmd.SetHelpTemplate(registry.HelpTemplate())
+	imageSearchCmd.SetUsageTemplate(registry.UsageTemplate())
+	imageSearchFlags := imageSearchCmd.Flags()
+	searchFlags(imageSearchFlags)
+}
+
+// searchFlags set the flags for the pull command.
+func searchFlags(flags *pflag.FlagSet) {
+	flags.StringSliceVarP(&searchOptions.Filters, "filter", "f", []string{}, "Filter output based on conditions provided (default [])")
+	flags.StringVar(&searchOptions.Format, "format", "", "Change the output format to a Go template")
+	flags.IntVar(&searchOptions.Limit, "limit", 0, "Limit the number of results")
+	flags.BoolVar(&searchOptions.NoTrunc, "no-trunc", false, "Do not truncate the output")
+	flags.StringVar(&searchOptions.Authfile, "authfile", buildahcli.GetDefaultAuthFile(), "Path of the authentication file. Use REGISTRY_AUTH_FILE environment variable to override")
+	flags.BoolVar(&searchOptions.TLSVerifyCLI, "tls-verify", true, "Require HTTPS and verify certificates when contacting registries")
+	if registry.IsRemote() {
+		_ = flags.MarkHidden("authfile")
+		_ = flags.MarkHidden("tls-verify")
+	}
+}
+
+// imageSearch implements the command for searching images.
+func imageSearch(cmd *cobra.Command, args []string) error {
+	searchTerm := ""
+	switch len(args) {
+	case 1:
+		searchTerm = args[0]
+	default:
+		return errors.Errorf("search requires exactly one argument")
+	}
```

Comment:
```suggestion
```
In command declaration: Args: cobra.ExactArgs(1) will take care of this.

Reply:
That's nice, thanks!